File: sdk/zig/examples/context.zig
const std = @import("std");
const lunaris = @import("lunaris");

export fn wmain(a: i32, b: i32) i32 {
    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
    defer arena.deinit();

    var ctx = lunaris.TaskContext.load(arena.allocator()) catch return a + b;
    defer ctx.deinit(arena.allocator());

    std.debug.print(
        "task={d} worker={s} caps={s}\n",
        .{ ctx.task_id, ctx.worker_version, ctx.host_capabilities_json },
    );

    return lunaris.simd.addChecked(a, b) catch a + b;
}

File: sdk/c/examples/context.c
#include "../lunaris.h"

#include <stdio.h>

int wmain(int a, int b) {
    lunaris_context_t ctx;
    lunaris_status_t status = lunaris_context_load(&ctx);

    if (status == LUNARIS_STATUS_OK) {
        printf(
            "task=%llu worker=%s caps=%s\n",
            (unsigned long long)ctx.task_id,
            ctx.worker_version,
            ctx.host_capabilities_json
        );
        lunaris_context_free(&ctx);
    }

    if (lunaris_has_capability("simd")) {
        return lunaris_simd_add_checked(a, b, NULL);
    }
    return a + b;
}
